plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -1050,6 +1058,13 @@ def exec_module(self, **kwargs):
                 differences = []
                 current_nics = []
                 results = vm_dict
+                current_osdisk = vm_dict['properties']['storageProfile']['osDisk']
+                current_ephemeral = current_osdisk.get('diffDiskSettings',None)
+
+                if self.ephemeral_os_disk and not current_ephemeral:
+                    self.fail('Ephemeral OS disk not updatable: virtual machine ephemeral OS disk is {0}'.format(self.ephemeral_os_disk))
```

Comment:
using self.ephemeral_os_disk is not None and self.ephemeral_os_disk != current_ephemeral
as the condition to judge whether self.-ephemeral_os_disk setting is to be updated.

Reply:
Ok! I will updated!